src/shadowbox/infrastructure/logging.ts

```diff
@@ -12,25 +12,45 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import * as path from 'path';
 
-function makeLogMessage(level: string, message: string): string {
+interface Callsite {
+  getLineNumber(): number;
+  getFileName(): string;
+}
+
+function getCallsite(): Callsite {
```

Comment:
This really needs some comments, e.g. is this trick V8-specific or is it a Node.js thing?

Reply:
Added a comment.